shell/platform/android/io/flutter/plugin/platform/PlatformViewWrapper.java

```diff
@@ -66,12 +68,39 @@ public void onFrameConsumed() {
         }
       };
 
+  private boolean shouldRecreateSurfaceForLowMemory = false;
+  private final TextureRegistry.OnLowMemoryListener lowMemoryListener =
+      new TextureRegistry.OnLowMemoryListener() {
+        @Override
+        public void onLowMemory(int level) {
+          // When a memory pressure warning is received and the level equal {@code
+          // ComponentCallbacks2.TRIM_MEMORY_COMPLETE}, the Android system release the underlying
+          // surface. If we continue to use the surface (e.g., call lockHardwareCanvas), a crash
+          // occurs, and we found that this crash appeared on Android10 and above.
+          //
+          // Here our workaround is to recreate the surface before using it.
```

Comment:
nit: also point to the issue this is fixing

Reply:
done.